### PR TITLE
compiler: restrict qualifier as property

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: devito/types/basic.py [hunk 1]
```diff
@@ -945,6 +945,10 @@ def name(self):
         """The name of the object."""
         return self._name
 
+    @property
+    def _C_type_qualifier(self):
```
Comment (georgebisbas) [inline devito/types/basic.py:949]: Should any docstring be added, or obvious?

**Comment by FabioLuporini** [comment]
I think it should be inferred from this instead https://github.com/devitocodes/devito/blob/44dc63b00360ce70ff93c0e19191148d27bdc36f/devito/tools/utils.py#L279

**Comment by georgebisbas** [comment]
> I think it should be inferred from this instead
> 
> https://github.com/devitocodes/devito/blob/44dc63b00360ce70ff93c0e19191148d27bdc36f/devito/tools/utils.py#L279

for  class IndexedData(IndexedBase) as well?

for AbstractFunction, ok

File: devito/types/basic.py (hunk 1 above)
Comment (mloubout) [inline devito/types/basic.py:949]: Any reason why we don't want it in `CodeSymbol` with the rest?

Comment (georgebisbas) [reply devito/types/basic.py:949]: right, thanks

**Comment by mloubout** [comment]
What happend here?

**Comment by georgebisbas** [comment]
> What happend here?

hell, I pushed from my fork! WIll cleanup...

**Comment by FabioLuporini** [comment]
the `unexpansion` branch probably has the machinery to do this right, but it's weeks away from merging state

**Comment by FabioLuporini** [comment]
closing as superseded by #2060

**Comment by FabioLuporini** [comment]
@georgebisbas you decide whether it's worth keeping the branch around or not

**Comment by georgebisbas** [comment]
I guess as soon as I rebase my devito-xdsl-fork on top of latest master I will let you know. I am fine to close now, thanks for the fix in your PR

**Pull request closed without merging**